### PR TITLE
Use privilege token in place of recovery approved token

### PR DESCRIPTION
**Pull request by kimlisa**
The changes will seem hectic if viewing changes all at once, so I tried to make each commit contained to a specific change

#### Description
Use privilege token in place of recovery approved token. Both tokens did the same thing as noted by @alex-kovoy

- Rename `ApproveAccountRecovery` to `VerifyAccountRecovery` to match the language we use for the front (and it seems more accurate)
- Delete `UserTokenTypeRecoveryApproved` and replace with `UserTokenTypePrivilege`

### CODE REVIEW

File: api/client/proto/authservice.proto [hunk 1]
```diff
@@ -917,20 +917,15 @@ message DeleteMFADeviceResponseAck {}
 
 // DeleteMFADeviceSyncRequest is a request to delete a MFA device (nonstream).
 message DeleteMFADeviceSyncRequest {
-    // TokenID is the ID of a user token that will be used to verify this request.
-    // Token types accepted are:
-    //   - Recovery approved token that is obtained with RPC ApproveAccountRecovery
-    //   - Privilege token that is obtained with RPC CreatePrivilegeToken
-    string TokenID = 1 [ (gogoproto.jsontag) = "token_id" ];
+    // PrivilegeTokenID is the ID of a privilege token that will be used to verify this request.
+    string PrivilegeTokenID = 1 [ (gogoproto.jsontag) = "privilege_token_id" ];
```
Comment (codingllama) [inline api/client/proto/authservice.proto:921]: Do you think it's worth renaming this one? I'm thinking we should keep the more generic "Token ID" name used throughout the API (with supporting documentation), so as we add/remove support for other tokens we keep the nomenclature consistent.

A possible exception would be places where we are absolutely sure that only a single token type will ever be accepted (RPCs exclusive to the account recovery flow, for example).

Applies elsewhere as well.

**Review by codingllama — APPROVED** [review]
LGTM.

Thanks for following up with this simplification/cleanup.

**Review by codingllama — APPROVED** [review]

**Review by klizhentas — APPROVED** [review]

**Comment by alex-kovoy** [comment]
> both tokens did the same thing as noted by @alex-kovoy

To clarify, because I was told that they both do the same thing I suggested to combine them but if they are different (in their intent and security scope), I would rather have a clear distinction between them.

**Comment by kimlisa** [comment]
closing in favor of https://github.com/gravitational/teleport/pull/8395

**Pull request closed without merging**